js/components/pastAndUpcomingEvents/EventLists.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import EventCard from '../EventCard';
+import Loading from '../Loading';
+import PropTypes from 'prop-types';
+
+const EventLists = ({ events, isFetchingMore, fetchMore, reachEnd, eventTime, showPastEvents }) => {
+
+  if (events.length === 0 && !isFetchingMore) {
+    return <p style={{marginLeft: "auto", marginRight: "auto"}}>Sorry, we could not find the match in {eventTime} events</p>
+  }
+
+  return (
+    <> 
+      {events.map((event) => (
+        <div className={`col-md-12 max-min-width event-${eventTime}`} key={event.id}>
+          <EventCard event={event} />
+        </div>
+      ))}
+
+      <div className="event-load-more">
+        { reachEnd ?
+        ( eventTime === "upcoming" && showPastEvents ? null : <p className="margin-5">Sorry, we reached the end with this search</p> ) : 
```

Comment:
We'll want to update the wording on this as it is syntactically odd. Try something along the lines of

> We have reached the end of results for the current filters

or

> No more results to show for current filters

Reply:
Updated the text